torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1531,13 +1531,12 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
   }
 }
 
-Stmt* TensorExprKernel::generateStmt(BackendType backendType) {
-  // Find all tensors we need to compute (including dependencies) and put them
-  // in a topological order
-  std::vector<Tensor*> tensors_to_compute =
-      findAllNeededTensors(tensorOutputs_);
-
-  torch::jit::tensorexpr::LoopNest l(tensorOutputs_, tensors_to_compute);
+Stmt* TensorExprKernel::generateStmt(BackendType backendType, Stmt* st) {
+  std::unordered_set<const Buf*> output_bufs;
+  for (auto t : tensorOutputs_) {
```

Comment:
Could we make replace `tensorOutputs_` with `bufOutputs_` in `TensorExprKernel` and collect Bufs instead of Tensors in the first place? That would get rid of this loop.

Reply:
Sure. I will do this cleanup as a followup.